core/services/offchainreporting/OffchainReportingAggregator.go

```diff
@@ -0,0 +1,233 @@
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/pkg/errors"
+)
+
+const (
+	OffchainReportingAggregatorName = "OffchainReportingAggregator"
+)
+
+var (
+	OffchainReportingAggregatorConfigSet = getConfigSetHash()
+
+	OffchainReportingAggregatorLogTypes = map[gethCommon.Hash]eth.Log{
+		OffchainReportingAggregatorConfigSet: &LogConfigSet{},
+	}
+)
+
+type (
+	LogConfigSet struct {
+		eth.GethRawLog
+		ConfigCount  uint64
+		Oracles      []gethCommon.Address
+		Transmitters []gethCommon.Address
+		Threshold    uint8
+		Config       []byte
+	}
+
+	OffchainReportingAggregator struct {
+		ethClient        eth.Client
+		configSetChan    chan offchainaggregator.OffchainAggregatorConfigSet
+		contractFilterer *offchainaggregator.OffchainAggregatorFilterer
+		contractCaller   *offchainaggregator.OffchainAggregatorCaller
+		contractAddress  gethCommon.Address
+		logBroadcaster   eth.LogBroadcaster
+
+		jobID       models.ID
+		transmitter Transmitter
+		contractABI abi.ABI
+	}
+
+	Transmitter interface {
+		CreateEthTransaction(ctx context.Context, toAddress gethCommon.Address, payload []byte) error
+		FromAddress() gethCommon.Address
+	}
+)
+
+var (
+	_ ocrtypes.ContractConfigTracker      = &OffchainReportingAggregator{}
+	_ ocrtypes.ContractTransmitter        = &OffchainReportingAggregator{}
+	_ ocrtypes.ContractConfigSubscription = &OffchainReportingAggregatorConfigSubscription{}
+)
+
+func NewOffchainReportingAggregator(address gethCommon.Address, ethClient eth.Client, logBroadcaster eth.LogBroadcaster, jobID models.ID, transmitter Transmitter) (o *OffchainReportingAggregator, err error) {
+	contractFilterer, err := offchainaggregator.NewOffchainAggregatorFilterer(address, ethClient)
+	if err != nil {
+		return o, err
+	}
+
+	contractCaller, err := offchainaggregator.NewOffchainAggregatorCaller(address, ethClient)
+	if err != nil {
+		return o, err
+	}
+
+	contractABI, err := abi.JSON(strings.NewReader(offchainaggregator.OffchainAggregatorABI))
+	if err != nil {
+		return o, err
+	}
+
+	return &OffchainReportingAggregator{
+		contractFilterer: contractFilterer,
+		contractCaller:   contractCaller,
+		contractAddress:  address,
+		logBroadcaster:   logBroadcaster,
+		ethClient:        ethClient,
+		configSetChan:    make(chan offchainaggregator.OffchainAggregatorConfigSet),
+		jobID:            jobID,
+		transmitter:      transmitter,
+		contractABI:      contractABI,
+	}, nil
+}
+
+func (ra *OffchainReportingAggregator) Transmit(ctx context.Context, report []byte, rs, ss [][32]byte, vs [32]byte) error {
+	payload, err := ra.contractABI.Pack("transmit", report, rs, ss, vs)
+	if err != nil {
+		return errors.Wrap(err, "abi.Pack failed")
+	}
+
+	return errors.Wrap(ra.transmitter.CreateEthTransaction(ctx, ra.contractAddress, payload), "failed to send Eth transaction")
+}
+
+func (ra *OffchainReportingAggregator) SubscribeToNewConfigs(ctx context.Context) (ocrtypes.ContractConfigSubscription, error) {
```

Comment:
context is ignored?

Reply:
It wouldn't do anything. Subscribing doesn't make any synchronous external calls.

Reply:
I see. Can we make it ```
_ context.Context
```
in that case?